### DISCUSSION TITLE
ManuallySyncingCommandExecutor saving errors

### DISCUSSION THREAD

**Discussion by daniil1112** [vladimir-bukhtoyarov/bucket4j#441] [General]
https://github.com/bucket4j/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/distributed/proxy/optimization/manual/ManuallySyncingCommandExecutor.java#L121 

Because of this line, in case of a short-term error by redis or another command executor, next command won't be executed successfully due to the preservation of CompletableFuture.fail

This behavior was intended as such?

**Comment by vladimir-bukhtoyarov** [comment]
> This behavior was intended as such?

No, it looks like a bug. I will release the fix at weekend.

**Reply by vladimir-bukhtoyarov** [reply]
@daniil1112 fixed in version 8.7.1
